File: backend/Models/Campaign.cs
namespace influenco.backend.Models;

public enum CampaignStatus
{
    Draft,
    OpenForApplications,
    Closed
}

public class Campaign : BaseEntity
{
    public Guid BrandId { get; set; }
    public Brand Brand { get; set; } = null!;

    public string Title { get; set; } = null!;
    public string? Description { get; set; }
    public string? Deliverables { get; set; }

    public decimal? Budget { get; set; }
    public DateTime? ApplicationDeadline { get; set; }
    public CampaignStatus Status { get; set; } = CampaignStatus.Draft;

    public List<string> Niches { get; set; } = new();
    public List<string> Platforms { get; set; } = new();
    public int? MinimumFollowers { get; set; }

    public int ApplicantsCount { get; set; } = 0;
}

File: frontend/src/api/types/campaign.ts
export type CampaignStatus = "Draft" | "OpenForApplications" | "Closed";

export interface Campaign {
    id: string;
    title: string;
    description?: string;
    deliverables?: string;
    budget?: number;
    applicationDeadline?: string;
    status: CampaignStatus;
    niches: string[];
    platforms: string[];
    minimumFollowers?: number;
    applicantsCount: number;
    createdAt: string;
}

export interface CampaignSummary {
    id: string;
    title: string;
    applicantsCount: number;
    applicationDeadline?: string;
    status: CampaignStatus;
}

export interface CampaignFormData {
    title: string;
    description?: string;
    deliverables?: string;
    budget?: number;
    applicationDeadline?: string;
    status: CampaignStatus;
    niches: string[];
    platforms: string[];
    minimumFollowers?: number;
}